lib/p2p/packets/Packet.ts

```diff
@@ -82,46 +78,26 @@ abstract class Packet<T = any> implements PacketInterface {
 
       if (bodyOrPacket) {
         this.body = bodyOrPacket;
-        this.header.hash = Packet.hash(bodyOrPacket);
       }
     }
   }
 
-  private static hash(value: any): string {
-    return MD5(stringify(value)).toString(CryptoJS.enc.Base64);
-  }
-
   public abstract serialize(): Uint8Array;
 
-  /**
-   * Verify the header hash against the packet body.
-   */
-  public verifyDataIntegrity(): boolean {
-    if (!this.body) {
-      return true;
-    }
-
-    if (!this.header.hash) {
-      return false;
-    }
-
-    return this.header.hash === Packet.hash(this.body);
+  public toJSON() {
+    return stringify({ header: this.header, body: this.body });
   }
 
   /**
    * Serialize this packet to binary Buffer.
    * @returns Buffer representation of the packet
    */
   public toRaw(): Buffer {
-    const msg = this.serialize();
-
-    const type = Buffer.alloc(1);
-    type.writeUInt8(this.type, 0, true);
-
-    const size = Buffer.allocUnsafe(4);
-    size.writeUInt32LE(msg.length, 0, true);
+    return Buffer.from(this.serialize().buffer as ArrayBuffer);
+  }
 
-    return Buffer.concat([type, size, Buffer.from(msg.buffer as ArrayBuffer)]);
+  public checksum(): Buffer {
```

Comment:
Also, for stylistic purposes we're not using the arrow function format for this and several other methods in this class, so it'd be worth changing. I think `checksum` might actually make the most sense as a getter.

Reply:
I remember having issues with the inheritance after using arrow functions, but I just checked and it's not the case. I'll change it (and `checksum` to a getter).

Reply:
I changed `checksum` to a property, since it's to be calculated only in the constructor.

Reply:
In regards to `checksum` - I'd rather not have a getter which is calculating stuff. Making it a property isn't good either, because it's supposed to be only for the wire protocol. So I ended up leaving it as a function.